go/engine/paperprovision.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package engine
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+)
+
+type PaperProvisionEngine struct {
+	libkb.Contextified
+	Username   string
+	DeviceName string
+	PaperKey   string
+	result     error
+	lks        *libkb.LKSec
+	User       *libkb.User
+}
+
+func NewPaperProvisionEngine(g *libkb.GlobalContext, Username string, DeviceName string, PaperKey string) *PaperProvisionEngine {
+	return &PaperProvisionEngine{
+		Contextified: libkb.NewContextified(g),
+		Username:     Username,
+		DeviceName:   DeviceName,
+		PaperKey:     PaperKey,
+	}
+}
+
+func (e *PaperProvisionEngine) Name() string {
+	return "PaperProvision"
+}
+
+func (e *PaperProvisionEngine) Prereqs() Prereqs {
+	return Prereqs{}
+}
+
+func (e *PaperProvisionEngine) RequiredUIs() []libkb.UIKind {
+	return []libkb.UIKind{
+		libkb.ProvisionUIKind,
+		libkb.LogUIKind,
+		libkb.SecretUIKind,
+		libkb.LoginUIKind,
+	}
+}
+
+func (e *PaperProvisionEngine) Run(ctx *Context) (err error) {
+	e.G().Log.Debug("+ PaperProvisionEngine Run")
```

Comment:
can you use `defer e.G().Trace("PaperProvisionEngine#Run", func() error { return err})()`?